cmd/kubeconfig.go

```diff
@@ -163,11 +164,26 @@ Note: A certificate once signed cannot be revoked for a particular user`,
 				if err != nil {
 					log.Fatal(err)
 				}
-				os.Stdout.Write(content)
+
+				clusterAPIAddress, err := getAPIAddress()
+				if err != nil {
+					return errors.Wrap(err, "failed to fetch cluster's API Address: %v.")
+				}
+				newContent := strings.Replace(string(content), "https://localhost:6443", clusterAPIAddress, -1)
+				os.Stdout.Write([]byte(newContent))
 			} else {
 				return errors.Errorf("failed to read admin config, is the control plane initialized on this node?")
 			}
 			return nil
 		},
 	}
 )
+
+func getAPIAddress() (string, error) {
+	clusterConfig, err := ConfigFromYaml(cfgFile)
+	if err != nil {
+		return "", err
+	}
+	address := clusterConfig.Spec.API.APIAddress()
+	return address, nil
```

Comment:
```suggestion
	return clusterConfig.Spec.API.APIAddress(), nil
```

Or does that not work?